src/lib/comms/include/sol-mqtt.h

```diff
@@ -0,0 +1,393 @@
+     *
+     * @param mqtt MQTT Object
+     * @param data User provided data
+     *
+     * @see sol_mqtt_connect
+     *
+     * Callback called when a subscribe request has been processed.
+     */
+    void (*subscribe) (struct sol_mqtt *mqtt, void *data);
+
+    /**
+     * @brief On unsubscribe callback
+     *
+     * @param mqtt MQTT Object
+     * @param data User provided data
+     *
+     * @see sol_mqtt_connect
+     *
+     * Callback called when a unsubscribe request has been processed.
+     */
+    void (*unsubscribe) (struct sol_mqtt *mqtt, void *data);
+};
+
+/**
+ * @brief Connect to a MQTT broker
+ *
+ * @param host The host address of the MQTT broker
+ * @param port The host port to connect to
+ * @param config Configuration and callbacks
+ * @param data User data provided to the callbacks
+ *
+ * @return New mqtt object on sucess, NULL otherwise
+ */
+struct sol_mqtt *sol_mqtt_connect(const char *host, int port, const struct sol_mqtt_config *config, void *data);
+
+/**
+ * @brief Reestablish the connection to the MQTT broker
+ *
+ * @param mqtt MQTT Object
+ *
+ * @return 0 on success, -EINVAL otherwise
+ */
+int sol_mqtt_reconnect(struct sol_mqtt *mqtt);
+
+/**
+ * @brief Disconnect from MQTT Broker
+ *
+ * @param mqtt MQTT Object;
+ *
+ * @see sol_mqtt_connect
+ *
+ * Terminate the connection to the broker and free the resources
+ * associated to the mqtt object.
+ */
+void sol_mqtt_disconnect(struct sol_mqtt *mqtt);
+
+/**
+ * @brief Get connection status
+ *
+ * @param MQTT Object
+ *
+ * @see sol_mqtt_conn_status
+ *
+ * @return sol_mqtt_conn_status code
+ */
+int sol_mqtt_get_connection_status(struct sol_mqtt *mqtt);
+
+/**
+ * @brief Send the Broker a message to be published in a given topic
+ *
+ * @param mqtt MQTT Object
+ * @param message Message to be published. The memory associated to this object should be handled by the caller.
+ *
+ * @return 0 on success, -EINVAL otherwise
+ */
+int sol_mqtt_publish(struct sol_mqtt *mqtt, struct sol_mqtt_message *message);
+
+/**
+ * @brief Ask the Broker to be subscribed to a given topic
+ *
+ * @params mqtt MQTT Object
+ * @params topic Topic to subscribe
+ * @param topic_len Topic length in bytes
+ * @param qos MQTT QOS that should be used by the subscribe message
+ *
+ * @return 0 on success, -EINVAL otherwise
+ */
+int sol_mqtt_subscribe(struct sol_mqtt *mqtt, char *topic, size_t topic_len, sol_mqtt_qos qos);
+
+/**
+ * @brief Create a new MQTT message
+ *
+ * @param topic Topic where the message will be published
+ * @param topic_len Length in bytes of the topic
+ * @param payload Payload of the message
+ * @param payload_len Length in bytes of the payload
+ * @param qos MQTT QOS that should be used when sending the message
+ * @param retain If the message should be retained by the broker
+ *
+ * @return New message object on success, NULL otherwise
+ */
+struct sol_mqtt_message *sol_mqtt_message_new(char *topic, size_t topic_len, void *payload,
```

Comment:
const void *payload? Also you could use the `const struct sol_buffer` to pass payload. If you make it clear it's const, then you're not doing any references of changes, just using both memory and size in a single structure.